egui/src/containers/panel.rs

```diff
@@ -241,10 +241,19 @@ impl SidePanel {
         let inner_response = frame.show(&mut panel_ui, |ui| {
             ui.set_min_height(ui.max_rect().height()); // Make sure the frame fills the full height
             ui.set_min_width(*width_range.start());
-            add_contents(ui)
+            let res = add_contents(ui);
+
+            // This is a hack to force the `Frame` to expand its background to fill the available
+            // space.
+            let mut rect = ui.min_rect();
+            rect.min -= Vec2::new(frame.margin.left, frame.margin.top);
+            rect.max += Vec2::new(frame.margin.right, frame.margin.bottom);
+            ui.allocate_space(ui.available_size());
```

Comment:
How does this work for left panels? Don't they just take up the entire width of the screen?

Reply:
Good question. It seems to work fine and I think this is because the  `panel_ui` given to the `SidePanel` has a its max_width set to the width of the panel, which initially is `default_width` and then `ui.min_rect().width()` for subsequent frames. I am not too familiar with the inner workings of the layout engine, but it seems to be this rect only grows if contents overflow somehow?

See [`Side::set_rect_width`](https://github.com/birktj/egui/blob/2d2345518ea8b65c7bd4695afaa8a912ee832c98/egui/src/containers/panel.rs#L53-L57), [the definition of `panel_rect`](https://github.com/birktj/egui/blob/2d2345518ea8b65c7bd4695afaa8a912ee832c98/egui/src/containers/panel.rs#L187-L196) and [`panel_ui`](https://github.com/birktj/egui/blob/2d2345518ea8b65c7bd4695afaa8a912ee832c98/egui/src/containers/panel.rs#L238-L241).